securedrop/tests/functional/test_source.py

```diff
@@ -158,9 +164,9 @@ def test_generate_codenames_in_multiple_tabs(self, sd_servers_v2, tor_browser_we
 
         # Then the submission fails and the user sees the corresponding flash message in Tab B
         self._assert_is_on_lookup_page(navigator)
-        notification = navigator.source_sees_flash_message()
+        notification = self._extract_flash_message_content(navigator)
```

Comment:
This logic is only needed for the tests in this specific file, and hence doesn't need to be in the navigator.